api/v1alpha1/envoyproxy_types.go

```diff
@@ -168,7 +170,7 @@ type FilterPosition struct {
 }
 
 // EnvoyFilter defines the type of Envoy HTTP filter.
-// +kubebuilder:validation:Enum=envoy.filters.http.cors;envoy.filters.http.ext_authz;envoy.filters.http.basic_authn;envoy.filters.http.oauth2;envoy.filters.http.jwt_authn;envoy.filters.http.fault;envoy.filters.http.local_ratelimit;envoy.filters.http.ratelimit;envoy.filters.http.wasm;envoy.filters.http.ext_proc;envoy.filters.http.rbac
+// +kubebuilder:validation:Enum=envoy.filters.http.health_check;envoy.filters.http.fault;envoy.filters.http.cors;envoy.filters.http.ext_authz;envoy.filters.http.basic_authn;envoy.filters.http.oauth2;envoy.filters.http.jwt_authn;envoy.filters.http.ext_proc;envoy.filters.http.wasm;envoy.filters.http.rbac;envoy.filters.http.local_ratelimit;envoy.filters.http.ratelimit
```

Comment:
should we remove https://github.com/envoyproxy/gateway/pull/3948/files#diff-741c6b9df53e59e3cace8bbd962d6dc2d30007bf69ccb875c0f60960fa73da4dL206-R211?

Reply:
But if people don`t know the last two filter and just want to insert the filter before router, the UX may be bad

Reply:
The default filter order is in the API doc so people know how to adjust them.